samples/Samples.MongoDB/Program.cs

```diff
@@ -109,5 +115,23 @@ public static async Task RunAsync(IMongoCollection<BsonDocument> collection, Bso
                 Console.WriteLine(allDocuments.FirstOrDefault());
             }
         }
+
+        public static void WireProtocolExecuteIntegrationTest(MongoClient client)
+        {
+            using (var syncScope = Tracer.Instance.StartActive("sync-calls-execute", serviceName: "Samples.MongoDB"))
+            {
+                var server = client.Cluster.SelectServer(new ServerSelector(), CancellationToken.None);
```

Comment:
It looks like `SelectServer` isn't available on all versions of MongoDb we test (it's not present on versions 2.0.2 and 2.1.1 for instance). Can you change this call to `SelectServerAsync` and then propagate the async calls all the way up to `Main`?

Reply:
Maybe then it's better to disable the test for those version, because I'm doing this is to hit the non async and non generic version of `void WireProtocol.Execute()`. WDYT?

Reply:
Yeah, I was thinking the same... I'll do the `#ifdef` change...

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/810/commits/2f75d5d7a6559de85b3026c5088b1576dfe0ad56